scripts/gen_vimdoc.py

```diff
@@ -1539,30 +1567,19 @@ def __bool__(self) -> bool:
                     (INCLUDE_DEPRECATED and self.deprecated_functions_text))
 
 
-def main(doxygen_config, args):
-    """Generates:
-
-    1. Vim :help docs
-    2. *.mpack files for use by API clients
-
-    Doxygen is called and configured through stdin.
-    """
-    for target in CONFIG:
-        if args.target is not None and target != args.target:
-            continue
-
-        config: Config = CONFIG[target]
+def generate(target: str, config: Config, doxygen_config, args):
```

Comment:
This function is now 190 LOC. Can we break this up?

Reply:
Of course! This work took the most important step for such a refactoring, so I can easily move out the inner function to outer scopes (previously everything was all in the for loop in `main()`...). I tried to minimize the diff even including whitespaces for doxygen stuffs, but we should make more agressive changes here. /cc @justinmk